apps/prairielearn/assets/scripts/lib/questionsTable.js

```diff
@@ -0,0 +1,194 @@
+    >`;
+    if (question.open_issue_count > 0) {
+      text += html`<a
+        class="badge badge-pill badge-danger ml-1"
+        href="${urlPrefix}/course_admin/issues?q=is%3Aopen+qid%3A${_.escape(question.qid)}"
+        >${question.open_issue_count}</a
+      >`;
+    }
+    return text.toString();
+  };
+
+  window.topicFormatter = function (topic, question) {
+    return html`<span class="badge color-${question.topic.color}"
+      >${_.escape(question.topic.name)}</span
+    >`.toString();
+  };
+
+  window.tagsFormatter = function (tags, question) {
+    return _.map(question.tags ?? [], (tag) =>
+      html`<span class="badge color-${tag.color}">${tag.name}</span>`.toString(),
+    ).join(' ');
+  };
+
+  window.versionFormatter = function (version, question) {
+    return html`<span class="badge color-${question.display_type === 'v3' ? 'green1' : 'red1'}"
+      >${_.escape(question.display_type)}</span
+    >`.toString();
+  };
+
+  window.topicSorter = function (topicA, topicB) {
+    return topicA.name.localeCompare(topicB.name);
+  };
+
+  window.genericFilterSearch = function (search, value) {
+    return $('<div>')
+      .html(value)
+      .find('.formatter-data')
+      .text()
+      .toUpperCase()
+      .includes(search.toUpperCase());
+  };
+
+  window.badgeFilterSearch = function (search, value) {
+    if (search === '(none)') {
+      return value === '';
+    }
+    var values = $('<div>')
+      .html(value)
+      .find('.badge')
+      .filter((i, elem) => $(elem).text().toUpperCase() === search.toUpperCase()).length;
+    return !!values;
+  };
+
+  let assessmentsByCourseInstanceFormatter = function (ci_id, question) {
+    var ci_assessments = _.filter(
+      question.assessments ?? [],
+      (assessment) => assessment.course_instance_id === ci_id,
+    );
+    return _.map(ci_assessments, (assessment) =>
+      html`<a
+        href="${plainUrlPrefix}/course_instance/${ci_id}/instructor/assessment/${assessment.assessment_id}"
+        class="badge color-${assessment.color} color-hover"
+        onclick="event.stopPropagation();"
+        ><span>${_.escape(assessment.label)}</span></a
+      >`.toString(),
+    ).join(' ');
+  };
+
+  let assessmentsByCourseInstanceList = function (ci_id) {
+    var data = $('#questionsTable').bootstrapTable('getData');
+    var assessments = _.filter(
+      _.flatten(_.map(data, (row) => row.assessments)),
+      (row) => row && row.course_instance_id === ci_id,
+    );
+    return _.assign(_.keyBy(_.map(assessments, (row) => row.label)), { '(None)': '(None)' });
+  };
+
+  course_instance_ids.forEach((courseInstanceId) => {
+    window[`assessments${courseInstanceId}List`] = function () {
+      return assessmentsByCourseInstanceList(courseInstanceId);
+    };
+
+    window[`assessments${courseInstanceId}Formatter`] = function (_, question) {
+      return assessmentsByCourseInstanceFormatter(courseInstanceId, question);
+    };
+  });
+
+  let tableSettings = {
```

Comment:
This can be a const. The object changes, but the variable still points to the same object.

Reply:
done